webapp/packages/plugin-data-viewer/src/DatabaseDataModel/IDatabaseDataSource.ts

```diff
@@ -80,6 +83,7 @@ export interface IDatabaseDataSource<TOptions, TResult extends IDatabaseDataResu
   setExecutionContext: (context: IConnectionExecutionContext | null) => this;
 
   retry: () => Promise<void>;
+  /** Allows to execute task after request or save action is done */
   runTask: <T>(task: () => Promise<T>) => Promise<T>;
```

Comment:
Allows to perform an asynchronous action on the data source, this action will wait previous action to finish and save or load requests. The data source will have a loading and disabled state while performing an action